### PR TITLE
Feature/checkboxes and strikes in todos

### PR DESCRIPTION
**Pull request by ghost**
Update is done property by clicking checkbox 
strikeline on done tasks

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -12,6 +12,20 @@ function App() {
   const [newTodoDescription, setNewTodoDescription] = useState("");
   const [loading, setLoading] = useState(true);
 
+  function ChangeTodo(url: string, id: string, isDone: boolean) {
```
Comment (alinauman02) [inline frontend/src/App.tsx]: Check naming convention.

File: frontend/src/components/TodoList.tsx [hunk 2]
```diff
@@ -6,14 +6,20 @@ import TodoListItem from "./TodoListItem";
 interface TodoListProps {
   todos: Todo[];
   deleteTodo: (id: string) => void;
+  checkTodo: (id: string, isDone: boolean) => void;
```
Comment (alinauman02) [inline frontend/src/components/TodoList.tsx]: When defining methods passed as `props`, its a convention to prefix their names with `on`, to make it clear to the user that the value passed should be a function.

Rename these props to `onDeleteTodo` and `onCheckTodo`,

**Review by alinauman02 — CHANGES_REQUESTED** [review]
Added some comments.

**Comment by ZunairaT** [comment]
When i run this branch there is an error being logged in the console. please review it.

**Review by alinauman02 — APPROVED** [review]

File: frontend/src/components/TodoListItem.tsx [hunk 3]
```diff
@@ -1,20 +1,42 @@
...
-export default function TodoListItem({ todo, deleteTodo }: TodoListItemProps) {
+export default function TodoListItem({
+  todo,
+  onChangeTodo,
+  onDeleteTodo,
+}: TodoListItemProps) {
+  const [check, setCheck] = useState(!todo.isDone);
+  const valueChangeHandler = (event: ChangeEvent<HTMLInputElement>) => {
+    onChangeTodo(todo.id, event.target.checked);
+    setCheck(!check);
```
Comment (alinauman02) [inline frontend/src/components/TodoListItem.tsx]: Again, this is the wrong way to update state if it is based on previous state. Furthermore, you do not need to store state for the check here in this component. With this approach, the array of Todos remains the same in the App component.

Remove the state here, and update the Todos array in the App component.

File: frontend/src/components/TodoListItem.tsx [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
+
   return (
     <li>
       <div className="list-item flex-container">
-        <p className="list-item-p">{todo.description}</p>
+        <input
+          className="check-box"
+          type="checkbox"
+          onChange={valueChangeHandler}
+          checked={!check}
```
Comment (alinauman02) [inline frontend/src/components/TodoListItem.tsx]: Use the `isDone` property of a Todo for `checked`, and also for rendering the description in the appropriate way below

**Review by alinauman02 — CHANGES_REQUESTED** [review]

**Review by ZunairaT — DISMISSED** [review]

File: frontend/src/App.tsx [hunk 5]
```diff
@@ -8,10 +8,36 @@ import { Todo } from "./models/todo.model";
 const urlString = "http://localhost:3001/todos/";
 
 function App() {
+  console.log("app rerendered");
```
Comment (ZunairaT) [inline frontend/src/App.tsx]: remove console log

**Review by ZunairaT — DISMISSED** [review]

File: frontend/src/components/TodoListItem.tsx (hunk 3 above)
Comment (alinauman02) [reply frontend/src/components/TodoListItem.tsx]: @Ejazhussainbaltistani You marked this conversation as resolved, but you did not make the changes

**Review by alinauman02 — APPROVED** [review]

**Review by ZunairaT — APPROVED** [review]

**Pull request merged**